### ISSUE TITLE
[Enhancement] Non-optimal regular expression doesn't check if pattern param is an array in preg_replace()

### ISSUE THREAD

**Issue by andreasschroth** [kalessil/phpinspectionsea#1065]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Enhancement                          |
| Plugin         | Php Inspections (EA Extended), 3.0.6.1 |
| Language level | PHP 7.2                        |

Current behaviour:
---

Here is some example code:

```php
preg_replace('/^[0-9]{2}/', '', '456');
preg_replace(['/^[0-9]{2}/'], '', '456');
```
The "non-optimal regular expression" inspection says correctly: "'[0-9]' can be replaced with '\d' (safe in non-unicode mode)." But only for the first line - the second line, where the passed `pattern` param to the `preg_replace()` is an array is not checked correctly.

Expected behaviour:
---

It's possible to pass either a single string as pattern to `preg_replace()` or an array of strings. Sure, in my very simplified example above it's useless to use the array, but there are many good use-cases for this. =)

I'd expect the inspection to also check all elements of the passed array as well.

**Comment by kalessil** [comment]
Marking as a bug, so it's on my short list.

**Comment by kalessil** [comment]
Fixed!

**Comment by andreasschroth** [comment]
Thank you!
